src/routes/AuthRedirect/AuthRedirect.tsx

```diff
@@ -0,0 +1,45 @@
+import queryString from "query-string";
+import {useEffect, useState} from "react";
+import Parse from "parse";
+import {API} from "api";
+import LoadingScreen from "components/LoadingScreen/LoadingScreen";
+import {useLocation} from "react-router";
+import ErrorPage from "components/ErrorPage/ErrorPage";
+
+function AuthRedirect() {
+  const [status, setStatus] = useState<{error?: string}>({});
+  const location = useLocation();
+  const params = queryString.parse(location.search);
+
+  useEffect(() => {
+    if (params.error) {
+      setStatus({error: params.error as string});
+    } else if (params.code && params.state) {
+      API.verifyGoogleSignIn(params.code as string).then((res) => {
+        const user = new Parse.User();
+        const authData = {
+          id: res.id,
+          id_token: res.idToken,
+          access_token: res.accessToken,
+        };
+
+        user.linkWith("google", {authData}).then(() => {
+          user.set("displayName", res.name);
+          user.save().then(() => {
+            window.location.href = decodeURI(params.state as string);
+          });
+        });
+      });
+    } else {
+      setStatus({error: "Not a valid entrypoint"});
+    }
+  }, [params.code, params.error, params.state]);
+
+  if (status.error) {
+    return <ErrorPage errorMessage="Oops! Unable to sign in." />;
```

Comment:
Wouldn't it be a good idea to show `status.error` on the error page?

Reply:
I really don't know what kind of information the error message might expose in the redirect from Google and whether it might contain some sensitive data or is targeted on technical users and contains some exceptions or stuff like that.
So I kept a generic message there and @miiho will design some generic error page for this, 404 errors and so on.